### PR TITLE
added the reset_ssh_session empty module and action plugin

### PR DESCRIPTION
**Pull request by hackndoes**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

module: reset_ssh_session
action plugin: reset_ssh_session

##### ANSIBLE VERSION

```
2.2.0.0
```

##### SUMMARY

Added new module (empty of code, only used for comfort and esthetics when calling the action
that will remove the ssh controllpersist handle file to force ansible to reconnect at that point in the execution enabling immediately apply of changes that require relogin to be used.

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure due to:

```
2017-02-14 13:14:11 == Missing __metaclass__ = type ==
2017-02-14 13:14:11 ./lib/ansible/plugins/action/reset_ssh_session.py
2017-02-14 13:14:11 
2017-02-14 13:14:11 == Missing from __future__ import (absolute_import, division, print_function) ==
2017-02-14 13:14:11 ./lib/ansible/plugins/action/reset_ssh_session.py
```

Here's an example of the missing boilerplate: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/copy.py#L19

File: lib/ansible/modules/system/reset_ssh_session.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/python
```
Comment (mattclay) [inline lib/ansible/modules/system/reset_ssh_session.py:1]: Modules should not have a shebang.

File: lib/ansible/modules/system/reset_ssh_session.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    control_persist_path: "{{ ansible_user_dir }}/.ansible/cp/ansible-ssh-{{ inventory_hostname }}-22-{{ ansible_user_id }}"
+'''
+
+RETURN = '''
+changed:
+  description: Specifies wether the control_persist_path points to a file and had it removed
+  returned: always
+  type: bool
+  sample: False
+'''
```
Comment (mattclay) [inline lib/ansible/modules/system/reset_ssh_session.py]: Missing newline at end of file.

File: lib/ansible/plugins/action/reset_ssh_session.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/python
```
Comment (mattclay) [inline lib/ansible/plugins/action/reset_ssh_session.py]: Plugins should not have a shebang.

**Comment by bcoca** [comment]
there is `meta: reset_connection`  that is more generic and should take care of this.

**Comment by sivel** [comment]
It looks like `reset_connection` is commented out for some reason.  We should probably get it working.  I prefer the `meta: reset_connection` approach better.

https://github.com/ansible/ansible/blob/db786fc21100737dde9633440072299d17a45350/lib/ansible/plugins/strategy/__init__.py#L905-L906

File: lib/ansible/modules/system/reset_ssh_session.py (hunk 1 above)
Comment (hackndoes) [reply lib/ansible/modules/system/reset_ssh_session.py:1]: I am looking in other modules under lib/ansible/system and they have a shabeng...

**Comment by hackndoes** [comment]
OK, I have this module only so I can call the action plugin using the module name directly ie. reset_ssh_session:

this module works fine with no code in it. It still fails when running the build so I tried to make it compatible with the standards you provide but now it also asks for a main method in the module (it doesn't need to run code... what should I do with that? and should I simply add the import module_utils?

File: lib/ansible/modules/system/reset_ssh_session.py (hunk 1 above)
Comment (mattclay) [reply lib/ansible/modules/system/reset_ssh_session.py:1]: Sorry, I was looking at too many PRs yesterday. You're right, it's plugins (not modules) which don't need a shebang.

**Comment by bcoca** [comment]
`meta: reset_connection` was 'temporarily' disabled .. and we forgot about it, re-enabling in #21494

File: lib/ansible/modules/system/reset_ssh_session.py (hunk 1 above)
Comment (mattclay) [reply lib/ansible/modules/system/reset_ssh_session.py:1]: However, modules which are only documentation (as this is), should not have a shebang. Here's an example of existing "module" which is documentation only:

https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/inventory/add_host.py

**Comment by hackndoes** [comment]
Ansible team reenabled unavailable feature - meta: reset_connection

**Pull request closed without merging**